src/decode.rs

```diff
@@ -4851,211 +4851,152 @@ unsafe fn decode_b(
-            == -(1 as libc::c_int)
-        {
-            let v: libc::c_int = dav1d_msac_decode_bools(
+            0
+        } as isize;
+
+        if *(t.cur_sb_cdef_idx_ptr).offset(idx) == -1 {
+            let v = dav1d_msac_decode_bools(
                 &mut ts.msac,
                 (*f.frame_hdr).cdef.n_bits as libc::c_uint,
-            ) as libc::c_int;
-            *(t.cur_sb_cdef_idx_ptr).offset(idx as isize) = v as int8_t;
-            if bw4 > 16 as libc::c_int {
-                *(t.cur_sb_cdef_idx_ptr)
-                    .offset((idx + 1 as libc::c_int) as isize) = v as int8_t;
+            ) as i8;
+
+            *(t.cur_sb_cdef_idx_ptr).offset(idx) = v;
+
+            if bw4 > 16 {
+                *(t.cur_sb_cdef_idx_ptr).offset(idx + 1) = v;
             }
-            if bh4 > 16 as libc::c_int {
-                *(t.cur_sb_cdef_idx_ptr)
-                    .offset((idx + 2 as libc::c_int) as isize) = v as int8_t;
+
+            if bh4 > 16 {
+                *(t.cur_sb_cdef_idx_ptr).offset(idx + 2) = v;
             }
-            if bw4 == 32 as libc::c_int && bh4 == 32 as libc::c_int {
-                *(t.cur_sb_cdef_idx_ptr)
-                    .offset((idx + 3 as libc::c_int) as isize) = v as int8_t;
+
+            if bw4 == 32 && bh4 == 32 {
+                *(t.cur_sb_cdef_idx_ptr).offset(idx + 3) = v;
             }
-            if DEBUG_BLOCK_INFO(f, t)
-            {
-                printf(
-                    b"Post-cdef_idx[%d]: r=%d\n\0" as *const u8 as *const libc::c_char,
-                    *t.cur_sb_cdef_idx_ptr as libc::c_int,
-                    ts.msac.rng,
-                );
+
+            if DEBUG_BLOCK_INFO(f, t) {
+                println!("Post-cdef_idx[{}]: r={}", *t.cur_sb_cdef_idx_ptr, ts.msac.rng);
             }
         }
     }
-    if t.bx & 31 as libc::c_int >> ((*f.seq_hdr).sb128 == 0) as libc::c_int == 0
-        && t.by & 31 as libc::c_int >> ((*f.seq_hdr).sb128 == 0) as libc::c_int
-            == 0
-    {
-        let prev_qidx: libc::c_int = ts.last_qidx;
-        let have_delta_q: libc::c_int = ((*f.frame_hdr).delta.q.present != 0
-            && (bs as libc::c_uint
-                != (if (*f.seq_hdr).sb128 != 0 {
-                    BS_128x128 as libc::c_int
-                } else {
-                    BS_64x64 as libc::c_int
-                }) as libc::c_uint || b.skip == 0)) as libc::c_int;
-        let mut prev_delta_lf: [int8_t; 4] = [0; 4];
-        memcpy(
-            prev_delta_lf.as_mut_ptr() as *mut libc::c_void,
-            (ts.last_delta_lf).as_mut_ptr() as *const libc::c_void,
-            4 as libc::c_int as libc::c_ulong,
-        );
-        if have_delta_q != 0 {
-            let mut delta_q: libc::c_int = dav1d_msac_decode_symbol_adapt4(
+
+    // delta-q/lf
+    let shfl = ((*f.seq_hdr).sb128 == 0) as libc::c_int;
+    if t.bx & (31 >> shfl) == 0 && t.by & (31 >> shfl) == 0 {
```

Comment:
```suggestion
    let not_sb128 = ((*f.seq_hdr).sb128 == 0) as libc::c_int;
    if t.bx & (31 >> not_sb128) == 0 && t.by & (31 >> not_sb128) == 0 {
```

I think a more descriptive name is better here.  In other places in the C code, `const int sb128 = f->seq_hdr->sb128` is used, so I think this way is closer.